maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/AbstractJavadocMojo.java

```diff
@@ -288,6 +289,8 @@
      */
     private static final float SINCE_JAVADOC_1_8 = 1.8f;
 
+    private static final float JAVA_VERSION_FLOAT = JavadocUtil.parseJavadocVersion(SystemUtils.JAVA_VERSION);
```

Comment:
I wouldn't use it for two reasons:

1. I don't like reflection personally, it is just ugly code
2. It work 9+ only

We can of course add more logic to Commons Lang parsing arbitrary strings from 6 to 9.